dcpy/git/github.py

```diff
@@ -0,0 +1,86 @@
+import os
+import requests
+from dateutil.parser import parse
+
+ORG = "NYCPlanning"
+PERSONAL_TOKEN = os.environ["GHP_TOKEN"]
+headers = {"Authorization": "Bearer %s" % PERSONAL_TOKEN}
+BASE_URL = f"https://api.github.com/repos/{ORG}"
+
+
+def parse_workflow(workflow):  ## json input
+    return {
+        "status": workflow["status"],
+        "conclusion": workflow["conclusion"],
+        "timestamp": parse(workflow["updated_at"], fuzzy=True),
+        "url": workflow["html_url"],
+    }
+
+
+def workflow_is_running(workflow: dict):
+    return workflow.get("status") in ["queued", "in_progress"]
+
+
+def get_default_branch(repo: str):
+    url = f"https://api.github.com/repos/nycplanning/{repo}"
+    response = requests.get(url).json()
+    return response["default_branch"]
+
+
+def get_branches(repo: str, branches_blacklist: list = None):
+    url = f"https://api.github.com/repos/nycplanning/{repo}/branches"
+    response = requests.get(url).json()
+    all_branches = [branch_info["name"] for branch_info in response]
+    if branches_blacklist is None:
+        branches_blacklist = []
+    return [b for b in all_branches if b not in branches_blacklist]
+
+
+def get_workflow(repo, name):
+    url = f"{BASE_URL}/{repo}/actions/workflows/{name}"
+    r = requests.get(url, headers=headers)
+    return r.json()
+
+
+def __get_workflow_runs_helper(url, params=None):
+    response = requests.get(url, headers=headers, params=params).json()
+    if "workflow_runs" in response:
+        return response["workflow_runs"]
+    else:
+        raise Exception(
+            f"Error retreiving workflow runs from Github. If error persists, contact Data Engineering. Github API response: {response}"
```

Comment:
This error was implemented like this because it displays when being used by GRU in their qaqc. Doesn't really belong here though, as this is a more general function and this error is a tad on the customer-focused side, so maybe should just throw an exception and handle the error display on the app side